From 4f605f13f9762bc640510aba46282a6883c3d66a Mon Sep 17 00:00:00 2001
From: Aihua Xu <aihuaxu@apache.org>
Date: Thu, 19 May 2016 20:11:01 -0400
Subject: [PATCH 0875/1164] CDH-48286 : HIVE-13798: Fix the unit test failure
 org.apache.hadoop.hive.cli.TestCliDriver.testCliDriver_ivyDownload
 (Aihua Xu, reviewed by Ashutosh Chauhan)

Change-Id: I22b029fdebea2a2c01f8e906429cd09a712da662
---
 conf/ivysettings.xml                               |    2 +-
 .../src/test/results/negative/cascade_dbdrop.q.out |    2 +-
 .../hadoop/hive/ql/processors/DfsProcessor.java    |    2 +-
 ql/src/test/queries/clientpositive/ivyDownload.q   |    6 +++++-
 .../test/results/clientpositive/ivyDownload.q.out  |    5 +++--
 5 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/conf/ivysettings.xml b/conf/ivysettings.xml
index bda842a..8a54f9b 100644
--- a/conf/ivysettings.xml
+++ b/conf/ivysettings.xml
@@ -30,7 +30,7 @@
       <ibiblio name="central" m2compatible="true"/>
       <!-- File resolver to add jars from the local system. -->
       <filesystem name="test" checkmodified="true">
-        <artifact pattern="/tmp/[module]-[revision](-[classifier]).jar" />
+        <artifact pattern="${test.tmp.dir}/[module]-[revision](-[classifier]).jar"/>
       </filesystem>
     </chain>
   </resolvers>
diff --git a/hbase-handler/src/test/results/negative/cascade_dbdrop.q.out b/hbase-handler/src/test/results/negative/cascade_dbdrop.q.out
index d2e0b81..f80f4a2 100644
--- a/hbase-handler/src/test/results/negative/cascade_dbdrop.q.out
+++ b/hbase-handler/src/test/results/negative/cascade_dbdrop.q.out
@@ -50,5 +50,5 @@ POSTHOOK: type: DROPDATABASE
 POSTHOOK: Input: database:hbasedb
 POSTHOOK: Output: database:hbasedb
 POSTHOOK: Output: hbasedb@hbase_table_0
-Command failed with exit code = 1
+#### A masked pattern was here ####
 Query returned non-zero code: 1, cause: null
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/processors/DfsProcessor.java b/ql/src/java/org/apache/hadoop/hive/ql/processors/DfsProcessor.java
index c3d5f81..23fcf26 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/processors/DfsProcessor.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/processors/DfsProcessor.java
@@ -90,7 +90,7 @@ public CommandProcessorResponse run(String command) {
 
       int ret = dfs.run(tokens);
       if (ret != 0) {
-        console.printError("Command failed with exit code = " + ret);
+        console.printError("Command " + command + " failed with exit code = " + ret);
       }
 
       System.setOut(oldOut);
diff --git a/ql/src/test/queries/clientpositive/ivyDownload.q b/ql/src/test/queries/clientpositive/ivyDownload.q
index 86ab648..2d3c25e 100644
--- a/ql/src/test/queries/clientpositive/ivyDownload.q
+++ b/ql/src/test/queries/clientpositive/ivyDownload.q
@@ -1,6 +1,8 @@
+dfs -cp ${system:hive.root}/contrib/target/hive-contrib-${system:hive.version}.jar ${system:test.tmp.dir}/udfexampleadd-1.0.jar;
+
 ADD JAR ivy://:udfexampleadd:1.0;
 
-CREATE TEMPORARY FUNCTION example_add AS 'UDFExampleAdd';
+CREATE TEMPORARY FUNCTION example_add AS 'org.apache.hadoop.hive.contrib.udf.example.UDFExampleAdd';
 
 EXPLAIN
 SELECT example_add(1, 2),
@@ -24,3 +26,5 @@ FROM src LIMIT 1;
 DROP TEMPORARY FUNCTION example_add;
 
 DELETE JAR ivy://:udfexampleadd:1.0;
+
+dfs -rm ${system:test.tmp.dir}/udfexampleadd-1.0.jar;
diff --git a/ql/src/test/results/clientpositive/ivyDownload.q.out b/ql/src/test/results/clientpositive/ivyDownload.q.out
index 23cc124..6bc9cb1 100644
--- a/ql/src/test/results/clientpositive/ivyDownload.q.out
+++ b/ql/src/test/results/clientpositive/ivyDownload.q.out
@@ -1,7 +1,7 @@
-PREHOOK: query: CREATE TEMPORARY FUNCTION example_add AS 'UDFExampleAdd'
+PREHOOK: query: CREATE TEMPORARY FUNCTION example_add AS 'org.apache.hadoop.hive.contrib.udf.example.UDFExampleAdd'
 PREHOOK: type: CREATEFUNCTION
 PREHOOK: Output: example_add
-POSTHOOK: query: CREATE TEMPORARY FUNCTION example_add AS 'UDFExampleAdd'
+POSTHOOK: query: CREATE TEMPORARY FUNCTION example_add AS 'org.apache.hadoop.hive.contrib.udf.example.UDFExampleAdd'
 POSTHOOK: type: CREATEFUNCTION
 POSTHOOK: Output: example_add
 PREHOOK: query: EXPLAIN
@@ -73,3 +73,4 @@ PREHOOK: Output: example_add
 POSTHOOK: query: DROP TEMPORARY FUNCTION example_add
 POSTHOOK: type: DROPFUNCTION
 POSTHOOK: Output: example_add
+#### A masked pattern was here ####
-- 
1.7.9.5

